h/services/oauth_validator.py

```diff
@@ -289,7 +293,13 @@ def validate_redirect_uri(self, client_id, redirect_uri, request, *args, **kwarg
 
         client = self.find_client(client_id)
         if client is not None:
-            return client.redirect_uri == redirect_uri
+            # Check that we match after a potentially templated redirect_uri
+            # has been templated out by our uri
+
+            return redirect_uri == render_url_template(
+                client.redirect_uri, example_url=redirect_uri
+            )
+
```

Comment:
Fix for bug 2, `confirm_redirect_uri()` did not work with templated redirect uris.